server/src/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -916,6 +916,9 @@ protected void updateSite2SiteVpnConnectionState(final List<DomainRouterVO> rout
                 }
                 continue;
             }
+            if (router.getIsRedundantRouter() && router.getRedundantState() != RedundantState.MASTER){
```

Comment:
after some more thought: You could be right. if the BACKUP is not running and goes through the previous inspection  AND has a attached vpn connection will be marked as disconnected. 
need to investigate a bit to see what _s2sVpnMgr.getConnectionsForRouter(router) would return for a backup router.